process/block/baseProcess.go

```diff
@@ -801,6 +793,50 @@ func (bp *baseProcessor) getLastCrossNotarizedHeadersForShard(shardID uint32) *b
 	return headerInfo
 }
 
+func (bp *baseProcessor) getLastSelfNotarizedHeaders() []bootstrapStorage.BootstrapHeaderInfo {
+	lastSelfNotarizedHeaders := make([]bootstrapStorage.BootstrapHeaderInfo, 0)
+
+	for shardID := uint32(0); shardID < bp.shardCoordinator.NumberOfShards(); shardID++ {
+		bootstrapHeaderInfo := bp.getLastSelfNotarizedHeadersForShard(shardID)
+		if bootstrapHeaderInfo != nil {
+			lastSelfNotarizedHeaders = append(lastSelfNotarizedHeaders, *bootstrapHeaderInfo)
+		}
+	}
+
+	bootstrapHeaderInfo := bp.getLastSelfNotarizedHeadersForShard(core.MetachainShardId)
+	if bootstrapHeaderInfo != nil {
+		lastSelfNotarizedHeaders = append(lastSelfNotarizedHeaders, *bootstrapHeaderInfo)
+	}
+
+	if len(lastSelfNotarizedHeaders) == 0 {
```

Comment:
no need for this if.

Reply:
We need this, as we don't want to marshal an empty slice with gogo protobuf. The result of this method is then added in bootstrapStorage.BootstrapData, marshalized and put it in bootstorer on disk.